### DISCUSSION TITLE
How to Perform Deposit and Withdrawal Using Safe API

### DISCUSSION THREAD

**Discussion by MWBlocky** [MixinNetwork/mixin#182] [Q&A]
I would like to perform the deposit and withdrawal process on the testnet.  
I have a few questions:

1. Is the current API I should be using the Safe API (https://developers.mixin.one/docs/api/safe-apis)?  
2. How do I create a test environment, and where can I get tokens for testing?  
3. I noticed that the Safe API documentation differs from the SDK for Node.js (https://github.com/MixinNetwork/bot-api-nodejs-client), for example in the response for pendingDeposits. Should Safe API be my only source in this case?  
4. The example for withdrawal seems outdated (https://github.com/MixinNetwork/bot-api-nodejs-client/blob/main/example/safe_withdraw.js), because in version "@mixin.dev/mixin-node-sdk": "^7.0.10", there is no method: `safe.fetchFee`.

**Comment by crossle** [comment]
1. Yes
2. Get some testing token please send message to Teamixin(@7000) on Mixin Messenger
3. Safe API is the correct form, maybe the node sdk has some old methods.
4. We'll fix the problem soon. Any problem when you used in your project feel free feedback.

Thanks your feedback.

**Reply by crossle** [reply]
node sdk v7.1.0 publish, please try again.
